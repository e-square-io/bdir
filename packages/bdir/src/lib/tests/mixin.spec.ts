import { renderSync } from 'sass';
import { readFileSync } from 'fs';

describe('mixins test', () => {
  function setup(source: string) {
    return renderSync({
      importer: (url) => {
        let filePath = `styles/${url.includes('utils') ? `_${url}` : url}.scss`;
        if (!process.cwd().includes('packages/bdir')) {
          filePath = `packages/bdir/${filePath}`;
        }

        return {
          contents: readFileSync(`./${filePath}`).toString(),
        };
      },
      data: `
        @import 'mixins';
        ${source}
        `,
    }).css.toString();
  }

  it('should create an encapsulated content', () => {
    const source = `
      h1 {
        @include encapsulation(true, '.test') {
          height: 10px;
        };
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for border end', () => {
    const source = `
      h1 {
        @include border-end(1px solid black);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for border-radius', () => {
    const source = `
      h1 {
        @include border-radius(0 2px 2px 0);
      }

      h2 {
        @include border-radius(4px 2px);
      }

      h3 {
        @include border-radius(4px 2px 5px);
      }

      h4 {
        @include border-radius(4px, true, false);
      }

      h5 {
        @include border-radius(4px 2px, false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for border-radius-start', () => {
    const source = `
      h1 {
        @include border-radius-start(2px);
      }

      h2 {
        @include border-radius-start(2px, 4px, false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for border-radius-end', () => {
    const source = `
      h1 {
        @include border-radius-end(2px);
      }

      h2 {
        @include border-radius-end(2px, 4px, false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for margin end', () => {
    const source = `
      h1 {
        @include margin-end(20px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for padding start', () => {
    const source = `
      h1 {
        @include padding-start(20px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for float', () => {
    const source = `
      div {
        @include float();
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for dir', () => {
    const source = `
      div {
        @include dir(true);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for start (left / right) positioning', () => {
    const source = `
      div {
        @include start(40px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for translating', () => {
    const source = `
      div {
        @include transformTranslate(5px, 5px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for scaling', () => {
    const source = `
      div {
        @include transformScale(5px, 5px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for mirroring', () => {
    const source = `
      div {
        @include mirror();
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for rotating', () => {
    const source = `
      div {
        @include transformRotate(45deg);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });
});
