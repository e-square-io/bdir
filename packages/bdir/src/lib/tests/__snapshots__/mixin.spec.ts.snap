// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixins test should create an encapsulated content 1`] = `
":host-context(.test) h1 {
  height: 10px;
}"
`;

exports[`mixins test should create an rtl ltr support for border end 1`] = `
":host-context(*[dir=ltr]) h1 {
  border-right: 1px solid black;
}
:host-context(*[dir=rtl]) h1 {
  border-left: 1px solid black;
}"
`;

exports[`mixins test should create an rtl ltr support for border-radius 1`] = `
":host-context(*[dir=ltr]) h1 {
  border-radius: 0 2px 2px 0;
}
:host-context(*[dir=rtl]) h1 {
  border-radius: 2px 0 0 2px;
}

:host-context(*[dir=ltr]) h2 {
  border-radius: 4px 0 0 2px;
}
:host-context(*[dir=rtl]) h2 {
  border-radius: 0 4px 2px 0;
}

h3 {
  border-radius: 4px 2px 5px;
}

:host-context(*[dir=ltr]) h4 {
  border-radius: 0 4px 4px 0;
}
:host-context(*[dir=rtl]) h4 {
  border-radius: 4px 0 0 4px;
}

*[dir=ltr] h5 {
  border-radius: 4px 0 0 2px;
}
*[dir=rtl] h5 {
  border-radius: 0 4px 2px 0;
}"
`;

exports[`mixins test should create an rtl ltr support for border-radius-end 1`] = `
":host-context(*[dir=ltr]) h1 {
  border-radius: 0 2px 2px 0;
}
:host-context(*[dir=rtl]) h1 {
  border-radius: 2px 0 0 2px;
}

*[dir=ltr] h2 {
  border-radius: 0 2px 4px 0;
}
*[dir=rtl] h2 {
  border-radius: 2px 0 0 4px;
}"
`;

exports[`mixins test should create an rtl ltr support for border-radius-start 1`] = `
":host-context(*[dir=ltr]) h1 {
  border-radius: 2px 0 0 2px;
}
:host-context(*[dir=rtl]) h1 {
  border-radius: 0 2px 2px 0;
}

*[dir=ltr] h2 {
  border-radius: 2px 0 0 4px;
}
*[dir=rtl] h2 {
  border-radius: 0 2px 4px 0;
}"
`;

exports[`mixins test should create an rtl ltr support for dir 1`] = `
":host-context(*[dir=ltr]) div {
  direction: left;
}
:host-context(*[dir=rtl]) div {
  direction: right;
}"
`;

exports[`mixins test should create an rtl ltr support for float 1`] = `
":host-context(*[dir=ltr]) div {
  float: left;
}
:host-context(*[dir=rtl]) div {
  float: right;
}"
`;

exports[`mixins test should create an rtl ltr support for margin end 1`] = `
":host-context(*[dir=ltr]) h1 {
  margin-right: 20px;
}
:host-context(*[dir=rtl]) h1 {
  margin-left: 20px;
}"
`;

exports[`mixins test should create an rtl ltr support for mirroring 1`] = `
":host-context(*[dir=ltr]) div {
  transform: scale(1, 1);
}
:host-context(*[dir=rtl]) div {
  transform: scale(-1, 1);
}"
`;

exports[`mixins test should create an rtl ltr support for padding start 1`] = `
":host-context(*[dir=ltr]) h1 {
  padding-left: 20px;
}
:host-context(*[dir=rtl]) h1 {
  padding-right: 20px;
}"
`;

exports[`mixins test should create an rtl ltr support for rotating 1`] = `
":host-context(*[dir=ltr]) div {
  transform: rotate(45deg);
}
:host-context(*[dir=rtl]) div {
  transform: rotate(-135deg);
}"
`;

exports[`mixins test should create an rtl ltr support for scaling 1`] = `
":host-context(*[dir=ltr]) div {
  transform: scale(5px, 5px);
}
:host-context(*[dir=rtl]) div {
  transform: scale(-5px, 5px);
}"
`;

exports[`mixins test should create an rtl ltr support for start (left / right) positioning 1`] = `
":host-context(*[dir=ltr]) div {
  left: 40px;
}
:host-context(*[dir=rtl]) div {
  right: 40px;
}"
`;

exports[`mixins test should create an rtl ltr support for translating 1`] = `
":host-context(*[dir=ltr]) div {
  transform: translate(5px, 5px);
}
:host-context(*[dir=rtl]) div {
  transform: translate(-5px, 5px);
}"
`;
